pkg/tsdb/cloudwatch/log_actions.go

```diff
@@ -0,0 +1,318 @@
+package cloudwatch
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs/cloudwatchlogsiface"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/tsdb"
+	"golang.org/x/sync/errgroup"
+)
+
+func (e *CloudWatchExecutor) executeLogActions(ctx context.Context, queryContext *tsdb.TsdbQuery) (*tsdb.Response, error) {
+	resultChan := make(chan *tsdb.QueryResult, len(queryContext.Queries))
+	eg, ectx := errgroup.WithContext(ctx)
+
+	for _, query := range queryContext.Queries {
+		query := query
+
+		eg.Go(func() error {
+			defer func() {
+				if err := recover(); err != nil {
```

Comment:
To be honest this error handling was copy/pasted from `executeTimeSeriesQuery` so @sunker might know better than me

Reply:
Fair enough. 

The cloudwatch metrics datasource have mostly been contributed to by external developers and not a high prio until a year ago. So before we started caring about it, the code quality was not great and it raised panics sometimes. Your code looks _much_ better so I think we should remove it for Cloudwatch logs. It's usually better to avoid defensive programming as it makes it harder to maintain code.